packages/mercury-case-analysis/src/rating.ts

```diff
@@ -0,0 +1,163 @@
+      `
+    ),
+  prompt_clarity: z
+    .object({
+      score: ratingSchema,
+      rationale: makeRationaleSchema("prompt clarity"),
+    })
+    .describe(
+      stripIndent`
+        How well-formulated and clear the prompt is.
+        A high score indicates that the prompt is clear and focused on a specific topic.
+        A low score indicates that the prompt is unclear or too broad.
+      `
+    ),
+  prompt_knowledge_assumption: z
+    .object({
+      score: ratingSchema,
+      rationale: makeRationaleSchema("prompt knowledge assumption"),
+    })
+    .describe(
+      stripIndent`
+        How much domain-specific knowledge might be required to effectively answer the prompt.
+        A high score indicates that the prompt is deeply technical or obscure.
+        A low score indicates that the prompt is relatively simple and straightforward.
+      `
+    ),
+  guidance: z
+    .string()
+    .optional()
+    .describe(
+      stripIndent`
+        A terse and direct suggestion for how to improve the prompt and/or expected response.
+        Only include this if ANY of the above scores <= 2.
+      `
+    ),
+});
+
+export type PromptResponseRating = z.infer<typeof promptResponseRatingSchema>;
+
+const schemaName = "prompt_response_rating";
+
+export type GenerateRating = (args: {
+  prompt: string;
+  response: string;
+}) => Promise<PromptResponseRating>;
+
+/**
+  Creates a function that, given the prompt and expected response pair, uses the
+  LLM to assess the quality on a variety of metrics and provide recommendations
+  for improvement.
+ */
+export function makeGenerateRating({
+  model,
+  styleGuide,
+}: {
+  model: LanguageModel;
+  styleGuide?: string;
+}): GenerateRating {
+  return async ({ prompt, response }): Promise<PromptResponseRating> => {
+    const result = await generateObject({
+      prompt: stripIndent`
+        Your task is to evaluate the quality of a technical chatbot prompt/response turn.
+        You will be given a user prompt and an expected answer for the prompt.
+        You will evaluate the quality of the prompt/response pair across multiple dimensions.
+        For each dimension, you will assign a score from 1 (poor) to 5 (excellent) as well as a rationale for your score.
+
+        Here are the dimensions you will evaluate:
+        ${YAML.stringify(
+          extractZodSchemaDescriptions(promptResponseRatingSchema),
+          {
+            indent: 4,
+          }
+        )}
+
+        ${
+          styleGuide
+            ? stripIndent`
+                Here is a style guide for the prompt/response pair:
+                <StyleGuide>
+                ${styleGuide}
+                </StyleGuide>
+              `
+            : ""
+        }
+
+        Now evaluate this prompt/expected response pair:
+
+        <Prompt>
+        ${prompt}
+        </Prompt>
+
+        <ExpectedResponse>
+        ${response}
+        </ExpectedResponse>
```

Comment:
id put this stuff in a user message

Reply:
For some reason I missed that `generateObject` also supports an array of messages - will change